devito/tools/data_structures.py

```diff
@@ -43,10 +43,15 @@ class EnrichedTuple(tuple, Pickable):
     A tuple with an arbitrary number of additional attributes.
     """
 
+    __rargs__ = ('*values',)
+    __rkwargs__ = ('getters',)
+
     def __new__(cls, *items, getters=None, **kwargs):
         obj = super().__new__(cls, items)
         obj.__dict__.update(kwargs)
-        obj._getters = OrderedDict(zip(getters or [], items))
+        if getters is not None or '_getters' not in kwargs:
```

Comment:
can `_getters

Reply:
Also this will be problematic if `getters` is `None` because `obj._getters` will not exist, that's what the `getters or ()` is for so that `obj._getters` always exists.

Reply:
I don't see why adding an `if` would be necessary?

Reply:
My reasoning was that otherwise, if reconstructed by something other than `Reconstructable._rebuild` (I believe there's some pickling system that uses `__getnewargs_ex__` instead, no?) then `kwargs['_getters']` will have the old value to be unpickled, but `getters` will equal `None`. Then if we don't do this check, the constructor will overwrite whatever the old value obtained from unpicking was, and instead overwrite with an empty dict

Reply:
Wait that was my old reasoning, I forgot I changed the way I did this so that's not accurate anymore. I'll take a look again in a minute

Reply:
Well, have found at least one case of extra kwargs being used `left` and `right` passed to a `DimensionTuple` constructor from `AbstractFunction`. I feel like in the long run that really be specialized rather than abusing the object `__dict__` this way but I'll leave it for now. 

Which means that we do need this check still, I think